cassandra/src/main/scala/filodb.cassandra/columnstore/CassandraColumnStore.scala

```diff
@@ -221,6 +221,61 @@ extends ColumnStore with CassandraChunkSource with StrictLogging {
     }
   }
 
+  /**
+   * Copy a range of partitionKey records to a target ColumnStore, for performing disaster recovery or
+   * backfills. This method can also be used to delete records, by specifying a ttl of zero.
+   * If the target is the same as the source, then this effectively deletes from the source.
+   *
+   * @param diskTimeToLiveSeconds pass zero to delete chunks
+   */
+  def copyOrDeletePartitionKeysByTimeRange(datasetRef: DatasetRef,
+                                           numOfShards: Int,
+                                           splits: Iterator[ScanSplit],
+                                           startTime: Long,
+                                           endTime: Long,
+                                           target: CassandraColumnStore,
+                                           targetDatasetRef: DatasetRef,
+                                           diskTimeToLiveSeconds: Int): Unit = {
+    def compareAndGet(sourceRec: PartKeyRecord, targetRec: PartKeyRecord): PartKeyRecord = {
+      // compare and get the oldest start time
+      val startTime =
+        if (sourceRec.startTime <= targetRec.startTime) sourceRec.startTime else targetRec.startTime
+      // compare and get the latest end time
+      val endTime =
+        if (sourceRec.endTime >= targetRec.endTime) sourceRec.endTime else targetRec.endTime
+
+      PartKeyRecord(sourceRec.partKey, startTime, endTime, None)
+    }
+
+    def copyRows(targetPartitionKeysTable: PartitionKeysTable, rows: Iterator[Row]) = {
+      for (row <- rows) {
+        val partKeyRecord = PartitionKeysTable.rowToPartKeyRecord(row)
+
+        if (diskTimeToLiveSeconds == 0) {
+          targetPartitionKeysTable.deletePartKey(partKeyRecord.partKey)
+        } else {
+          targetPartitionKeysTable.readPartKey(partKeyRecord.partKey) match {
+            case Some(targetPkr) =>
+              targetPartitionKeysTable.writePartKey(compareAndGet(partKeyRecord, targetPkr), diskTimeToLiveSeconds)
```

Comment:
We should also write it to `PartitionKeysByUpdateTimeTable` so that the downsampler job can pick it up. This should not be done when migrating the downsample keyspace.